### PR TITLE
[WIP] Fetched channel names to display in messages

### PR DESCRIPTION
**Pull request by camachom**
This PR is for issue #1 - attempting at displaying channel names. This is not a final PR: the idea is to hopefully get some feedback and make sure I'm in the right track.

This was the approach taken:
1. extract all the channel-id/channel-names from the messages
2. merge step-1 with all the messages from the channels query
3. if any channel name is missing at that point, query for it.
4. replace channel-id with channel-name

I tried to come up with a more specific regex than `#"(?<=<#).*?(?=>)"` that had groupings for `id` and `name`, but could not successfully do so. It would be nice to avoid having to split by `|`.

### CODE REVIEW

File: src/clj/clojurians_log/db/queries.clj [hunk 1]
```diff
@@ -64,4 +64,14 @@
          [?user :user/slack-id ?id]
          [?user :user/name ?username]]
        db
+       names))
+
+(defn channel-names
+  [db names]
```
Comment (Odie) [inline src/clj/clojurians_log/db/queries.clj]: The parameter `names` seems a bit misleading here. I think it'd be clearer if it's named something like `ids` or `channel-ids`.

File: src/clj/clojurians_log/slack_messages.clj [hunk 2]
```diff
@@ -104,9 +154,139 @@
 
 (defn message->text
   "Convert Slack markup to plain text."
-  [message usernames]
+  [message usernames channel-names]
```
Comment (Odie) [inline src/clj/clojurians_log/slack_messages.clj]: The function signature change causes an exception here:
https://github.com/clojureverse/clojurians-log-app/blob/70555154e393b30a3efb8a4dfe4cb0e2a7006a88/src/clj/clojurians_log/views.clj#L52

This probably just tells we need some tests to access the various routes to make sure it at least it doesn't throw an exception.

File: project.clj [hunk 3]
```diff
@@ -56,7 +56,7 @@
 
   :cljsbuild {:builds
               [{:id "app"
-                :source-paths ["src/cljs" "src/cljc" "dev"]
+                :source-paths ["src/cljs" "src/cljc" "src/clj" "dev"]
```
Comment (plexus) [inline project.clj]: Why is this needed?

File: src/clj/clojurians_log/routes.clj [hunk 4]
```diff
@@ -10,7 +10,8 @@
             [compojure.core :refer [GET routes]]
             [compojure.route :refer [resources]]
             [datomic.api :as d]
-            [ring.util.response :refer [response]]))
+            [ring.util.response :refer [response]]
+            [clojure.pprint :as pp]))
```
Comment (plexus) [inline src/clj/clojurians_log/routes.clj]: leftover from debugging?

File: src/clj/clojurians_log/slack_messages.clj [hunk 5]
```diff
@@ -1,16 +1,65 @@
 (ns clojurians-log.slack-messages
   (:require [clojurians-log.message-parser :as mp]
             [clojure.string :as str]
-            [hiccup2.core :as hiccup]))
+            [hiccup2.core :as hiccup]
+            [clojure.pprint :as pp]))
```
Comment (plexus) [inline src/clj/clojurians_log/slack_messages.clj]: can probably be removed again.

File: src/clj/clojurians_log/slack_messages.clj [hunk 6]
```diff
@@ -104,9 +154,139 @@
...
   (->> (-> message
            (mp/parse)
-           (replace-ids-names usernames))
+           (replace-user-ids-names usernames)
+           (replace-channel-ids-names channel-names))
        (map segment->text)
        (apply str)))
+
+
+(def test-response
```
Comment (plexus) [inline src/clj/clojurians_log/slack_messages.clj]: This doesn't seem to be used, it also looks a bit out of place here? Maybe move it to a `repl/` namespace?

File: src/clj/clojurians_log/views.clj [hunk 7]
```diff
@@ -3,7 +3,8 @@
             [cemerick.url :refer [url]]
             [clojurians-log.time-util :as cl.tu]
             [clojure.string :as str]
-            [clojurians-log.slack-messages :as slack-messages]))
+            [clojurians-log.slack-messages :as slack-messages]
+            [clojure.pprint :as pp]))
```
Comment (plexus) [inline src/clj/clojurians_log/views.clj]: Another pprint :)

**Comment by plexus** [comment]
Hi @camachom, I need to have a better look but I think you're on the right track. Note that both channels and users can optionally have the actual name in there, e.g. `<@U14JKFW|plexus>` , so maybe they can share some logic?

It seems the build failed, could you have a look what's up there?

It seems there aren't many tests yet for this part. Instead of having a single test that tries to hit every case, consider adding individual focused test cases. If you look back at the original check list for this parses, then every item on that list should really be its own test (or `(testing ,,,)` block).

File: src/clj/clojurians_log/slack_messages.clj (hunk 2 above)
Comment (Odie) [reply src/clj/clojurians_log/slack_messages.clj]: Btw, #31 exercises all currently known route types. So this kind of issue will be caught once that's been merge. 😄

**Comment by camachom** [comment]
Thanks for the great feedback @plexus @Odie! I will try to make my WIP PRs a little cleaner going forward.

I will wait to see what happens with PR #34, and then work on the individual test and integrating the user/channel naming code based on that.

**Comment by plexus** [comment]
@camachom I'm closing this as quite a lot has happened since this PR was last touched. Thanks a lot for the help!
